esp/platform/application_config.cpp

```diff
@@ -359,6 +372,27 @@ void setLDAPSecurityInWSAccess(IPropertyTree *legacyEsp, IPropertyTree *legacyLd
         addBindingToServiceResource(wsAccessService, binding.queryProp("@name"), "WsSMC",
             binding.getPropInt("@port"), baseDN, workunitsBaseDN);
     }
+
+    //Now, setLDAPSecurity for the esp applications other than eclwatch.
+    char sepchar = getPathSepChar(hpccBuildInfo.componentDir);
+    Owned<IPropertyTreeIterator> services = getGlobalConfigSP()->getElements("services[@class='esp'][@public='true']");
+    ForEach(*services)
+    {
+        IPropertyTree &service = services->query();
+        const char *type = service.queryProp("@type");
+        if (strieq(type, "eclwatch"))
+            continue;
+
+        StringBuffer path(hpccBuildInfo.componentDir);
+        addPathSepChar(path, sepchar).append("applications").append(sepchar).append(type).append(sepchar);
+        path.append("ldap_authorization_map.yaml");
+        Owned<IPropertyTree> authTree = createPTreeFromYAMLFile(path);
```

Comment:
What happens if this file doesn't exist?  The pod/service with the bad application name probably didn't start, but should EclWatch run anyway?

Reply:
Ok, yes, I see that if authTree is null getLDAPBaseDN simply returns null as well.  It's good that you are now logging the misconfiguration.